server-common/src/main/java/org/apache/kafka/server/group/share/Persister.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.group.share;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This interface introduces methods which can be used by callers to interact with the
+ * persistence impl responsible for storing share group/partition states.
+ * For KIP-932, the persistence impl would be a share coordinator which will store information in
+ * an internal topic. But this allows for other variations as well.
+ */
+@InterfaceStability.Evolving
+public interface Persister {
+  /**
+   * The InitializeShareGroupState API is used to initialize the share-partition state.
+   *
+   * @param request InitializeShareGroupStateParameters
+   * @return InitializeShareGroupStateResult
+   */
+  CompletableFuture<InitializeShareGroupStateResult> initializeShareGroupState(InitializeShareGroupStateParameters request);
```

Comment:
nit: do we need `shareGroup` in every method or they could be `initializeState`, etc. as the class itself belongs to package `....group.share`?